src/Relogify/AddResult.fs

```diff
@@ -1,48 +1,125 @@
-      OpponentName = "Player2"
-      OwnPoints = 0
-      OpponentPoints = 0
-      CanAddResult = false
-      ExtraTime = false }
+    { resultModel =
+        { OwnPoints = 0
+          OpponentPoints = 0
+          ExtraTime = false }
+      state = EditingResult }
 
 type Msg =
     | SetOwnPoints of int
     | SetOpponentPoints of int
     | ToggleExtraTime
+    | AddResultInitiated
+    | ResultAddedSuccess
+    | ResultAddedError of string
+
+type AddResultModel =
+    { CommunityName: string
+      PlayerName: string
+      OpponentName: string
+      PlayerPoints: int
+      OpponentPoints: int
+      ExtraTime: bool }
+
+type CmdMsg =
+    | Noop
+    | AddResultCmdMsg of AddResultModel
+
+let rand = Random()
 
-type CmdMsg = Noop
+let addResultCmd (resultModel: AddResultModel) =
+    async {
+        // replace getPlayersOperation with addResultOperation
+        do! Async.Sleep 1000
+        let! result = getPlayersOperation.AsyncRun(runtimeContext, resultModel.OpponentName) // TODO: Send in the real data
+        return
+            match result.Data with // TODO: result.Errors
+            | Some _ -> // ResultAddedSuccess // TODO: Uncomment this and remove the randomness
+                if rand.Next(2) = 0 then
+                    ResultAddedError "Failed to add result, please check your internet connection and try again"
+                else
+                    ResultAddedSuccess
+            | None -> ResultAddedError "Failed to add result, please check your internet connection and try again"
+    }
+    |> Cmd.ofAsyncMsg
 
 let mapCommands: CmdMsg -> Cmd<Msg> =
     function
     | Noop -> Cmd.none
+    | AddResultCmdMsg resultModel -> addResultCmd resultModel
 
-let update model msg: Model * CmdMsg list =
+let toAddResultModel (playerName: string) (opponentName: string) (communityName: string) (resultModel: ResultModel): AddResultModel =
+    { PlayerName = playerName
+      OpponentName = opponentName
+      CommunityName = communityName
+      PlayerPoints = resultModel.OwnPoints
+      OpponentPoints = resultModel.OpponentPoints
+      ExtraTime = resultModel.ExtraTime }
+
+let updateResultModel (resultModel: ResultModel) (msg: Msg): ResultModel =
     match msg with
-    | SetOwnPoints newOwnPoints -> { model with OwnPoints = newOwnPoints }, []
-    | SetOpponentPoints newOpponentPoints -> { model with OpponentPoints = newOpponentPoints }, []
-    | ToggleExtraTime -> { model with ExtraTime = not model.ExtraTime }, []
+    | SetOwnPoints newOwnPoints -> { resultModel with OpponentPoints = newOwnPoints }
+    | SetOpponentPoints newOpponentPoints -> { resultModel with OpponentPoints = newOpponentPoints }
+    | ToggleExtraTime -> { resultModel with ExtraTime = not resultModel.ExtraTime }
+    | _ -> resultModel
+
+let update (model: Model) (msg: Msg) (communityName: string) (ownName: string) (opponentName: string): Model * CmdMsg list =
+    match model.state, msg with
+    | (EditingResult, SetOwnPoints _)
```

Comment:
Maybe nested `match`?
```
match model.state with
| EditingResult | ErrorAddingResult ->
    match msg with
    | SetOwnPoints _ | SetOpponentPoints _ | ToggleExtraTime ->
    { model with state = EditingResult; resultModel = updateResultModel model.resultModel msg }, []
```

Reply:
Now that I've re-read the code, I can actually keep it flat, but always set state to ``EditingResult` at the end.
The idea with keeping it flat is to keep the pattern of matching on `State` and `Transition`, so that it's clear what states allow what transitions.